pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -598,6 +666,36 @@ func (p *Pod) equivalentToWorkload(wl *kueue.Workload, jobPodSets []kueue.PodSet
 	return true
 }
 
+func (p *Pod) ReclaimablePods() ([]kueue.ReclaimablePod, error) {
+	if p.isGroup {
+		var result []kueue.ReclaimablePod
+		for _, pod := range p.list.Items {
+			if pod.Status.Phase == corev1.PodSucceeded || (p.isUnretriableGroup() && !podActive(&pod)) {
+				roleHash, err := getRoleHash(pod)
+				if err != nil {
+					return nil, err
+				}
+
+				roleFound := false
+				for i := range result {
+					if result[i].Name == roleHash {
+						result[i].Count++
```

Comment:
Let's say the initial pod group size is 10, and we created 20 replacement pods after many failures.
At this point, the user adds a pod with the annotation. In that case, this function is returning more reclaimable pods than the group size.